tool/cmd/kitex/versions/version_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package versions
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cloudwego/kitex/internal/test"
+)
+
+func Test_newVersion(t *testing.T) {
+	testcases := []struct {
+		verStr        string
+		parts         []int
+		preRelease    string
+		buildMetadata string
+		expectErr     bool
+	}{
+		{
+			verStr: "v1.0.0",
+			parts:  []int{1, 0, 0},
+		},
+		{
+			verStr: "v12.34.56",
+			parts:  []int{12, 34, 56},
+		},
+		{
+			verStr:        "v1.0.0-rc1.0.12345678901234-abcd12345678+build.1.0.0",
+			parts:         []int{1, 0, 0},
+			preRelease:    "rc1.0.12345678901234-abcd12345678",
+			buildMetadata: "build.1.0.0",
+		},
+		{
+			verStr:     "v1.0.0-12345678901234-abcd12345678",
+			parts:      []int{1, 0, 0},
+			preRelease: "12345678901234-abcd12345678",
+		},
+		{
+			verStr:    "1.0.0",
+			expectErr: true,
+		},
+	}
+	for _, tc := range testcases {
+		ver, err := newVersion(tc.verStr)
+		if !tc.expectErr {
+			test.Assert(t, err == nil, err)
+			test.Assertf(t, reflect.DeepEqual(tc.parts, ver.parts), "want %v, got %v", tc.parts, ver.parts)
+		} else {
+			test.Assert(t, err != nil)
+		}
+
+	}
+}
+
+func TestVersion_lessThan(t *testing.T) {
+	testcases := []struct {
+		base     string
+		other    string
+		expected bool
+	}{
```

Comment:
Test case coverage could be improved, such as 

- ("v1.0.0", "v1.0.0")
- ("v0.1.2", "v0.2.1")

Reply:
done.